
# New isolate genomes and global marine metagenomes resolve ecologically relevant units of SAR11


New isolate genomes and global marine metagenomes resolve ecologically relevant units of SAR11
Kelle C. Freel, Sarah J. Tucker, Evan B. Freel, Ulrich Stingl, Stephen J. Giovannoni, A. Murat Eren, Michael S. Rapp√©
bioRxiv 2024.12.24.630191; doi: https://doi.org/10.1101/2024.12.24.630191

The bacterial order Pelagibacterales (SAR11) is widely distributed across the global surface ocean, where its activities are integral to the marine carbon cycle. High-quality genomes from isolates that can be propagated and phenotyped are needed to unify perspectives on the ecology and evolution of this complex group. Here, we increase the number of complete SAR11 isolate genomes threefold by describing 81 new SAR11 strains from coastal and offshore surface seawater of the tropical Pacific Ocean. Comprehensive analyses of genomes and their spatiotemporal distributions supports the existence of 29 monophyletic, discrete Pelagibacterales ecotypes that we define as genera. The spatiotemporal distributions of genomes within genera were correlated at fine scales with variation in ecologically-relevant gene content, supporting generic assignments and providing indications of speciation. We propose that complex SAR11 populations can be cast into a hierarchical system of classification for SAR11 populations that is meaningfully correlated with evolution and ecology, providing a valid and utilitarian systematic nomenclature for this clade.


### Data and materials availability
The assembled sequence data for genomes reported here including raw sequencing reads are available on NCBI associated with BioProject PRJNA1170004, and also available at FigShare at https://doi.org/10.6084/m9.figshare.28087454.v1 in addition to full length 16S rRNA gene sequences (10.6084/m9.figshare.29983285).

### Reproducible SAR11 phylogenomics workflow 
The URL https://merenlab.org/data/sar11-phylogenomics/ serves a reproducible bioinformatics workflow with all the genomes used in our study to reproduce our phylogenomic tree or extend it with new genomes.

### This directory has the code used for the analyses in the manuscript 
