---
title: "02_Phylogeny_268"
author: "Kelle"
date: "2024-10-24"
output: html_document
---

# Initial phylogeny

### Prepare the external-genomes.txt file

```{r}
# Remove the identified IV and V genomes from the "external_genomes_504_set_outgroup_v3.txt" file

# First read in the appropriate old file
external_genomes_504_set_outgroup_v3 <- read.delim("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/data/October24/phylogeny_prep/external_genomes_504_set_outgroup_v3.txt", stringsAsFactors = FALSE)

IV_V_remove <- read.delim("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/data/October24/phylogeny_prep/IV_and_V_remove_481.txt", stringsAsFactors = FALSE)

# This is one way to remove, BUT the names in the columns must be IDENTICAL, and I realized something was wrong so couldn't use this
# external_genomes_484_set_outgroup_v1 <- anti_join(external_genomes_504_set_outgroup_v3, IV_V_remove, by = c("name"))

# This is another way to remove, if only some of the characters are the same
external_genomes_484_set_outgroup_v1 <- external_genomes_504_set_outgroup_v3 %>%
  filter(!sapply(name, function(x) any(str_starts(x, IV_V_remove$name))))

write.csv(external_genomes_481_set_outgroup_v1, file = "/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/data/October24/phylogeny_prep/external_genomes_481_set_outgroup_v1.csv")

# 

```

### Generate a phylogeny using 481 SAR11 genomes and 10 outgroup genomes 

```{bash}
# Make an external-genomes.txt file that has all of the genomes and the paths to the contigs.db of interest

cd /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_504_outgroup_23oct24

anvi-get-sequences-for-hmm-hits --external-genomes external_genomes_481_set_outgroup_v1.csv \
                                -o concatenated_genes_v2_excl_IV_V/concatenated-best_alphaproteobacteria165.fa \
                                --hmm-sources Alphaproteobacterial_SCGs \
                                --gene-names acpS,acyl_carrier_protein,afg1_family_ATPase,alpha_beta_hydrolase_fold,ApbC,aspS,ATP_synthase_F0_sub_a,atpG,Beta_lactamase_like_protein,BolA_family_protein,ccmC,ccmE,cell_division_trigger_factor,coaD,coaE,copper_homeostasis,cytidylate_kinase,cytochrome_oxidase_assembly,dapD,delta_AAD,DNA_polymerase_I,DnaA,dnaA,dut_deoxyuridine_TN,eno_hydratase,exonuclease,fabZ,folylpolyglutamate_synth,fpg,frr,gatB,gidA,gidB,glyS,gpsA,grla,GrpE,grxC,hemE,heme_chaperone,Holliday_junction_resolvase_like,HslV,hydrolase_TatD_family,hypothetical_protein_01,hypothetical_protein_02,hypothetical_protein_03,hypothetical_protein_04,hypothetical_protein_05,hypothetical_protein_06,hypothetical_protein_07,hypothetical_protein_08,hypothetical_protein_09,hypothetical_protein_12,hypothetical_protein_14,infC,iojap_protein_family,ksgA,lipoate_protein_ligase_B,lipoyl_synthase,maf,malate_dehydrogenase,mesJ,metallo_beta_lactamase,methylase,miaA,mnmA,MraW,murF,murG,nifR3_TIM,NifU,nucleoside_diphosphate_kinase,nuoJ,nuoK2,nusA,nusG,parvulin_like_PPI,PDT,peptidyl_tRNA_hydrolase,pgk_phosphoglycerate_kinase,pgsA,phage_SPO1_related,pheT,phospho_N_APT,plsX,polyA_polymerase,porphobilinogen_deaminase,protein_S6,putative_DNA_binding,putative_zinc_metallo,pyrG,quinone_oxi,recA,recombination_protein_RecR,Ribonuclease_3,ribonuclease_E,ribosomal_protein_L1,ribosomal_protein_L11,ribosomal_protein_L13,ribosomal_protein_L14,ribosomal_protein_L15,ribosomal_protein_L17,ribosomal_protein_L18,ribosomal_protein_L19,ribosomal_protein_L2,ribosomal_protein_L21,ribosomal_protein_L22,ribosomal_protein_L23,ribosomal_protein_L24,ribosomal_protein_L27,ribosomal_protein_L28,ribosomal_protein_L3,ribosomal_protein_L4,ribosomal_protein_L5,ribosomal_protein_L6,ribosomal_protein_L9,ribosomal_protein_S10,ribosomal_protein_S11,ribosomal_protein_S12,ribosomal_protein_S13,ribosomal_protein_S14,ribosomal_protein_S15,ribosomal_protein_S16,ribosomal_protein_S17,ribosomal_protein_S18,ribosomal_protein_S19,ribosomal_protein_S2,ribosomal_protein_S20,ribosomal_protein_S3,ribosomal_protein_S4,ribosomal_protein_S5,ribosomal_protein_S7,ribosomal_protein_S8,ribosomal_protein_S9,Ribosome_binding_factor_A,RNA_polymerase_sub_a,rnhB,rplT,rrm1,rRNA_LS_methyltransferase,sdhC_556_sub,secY,serS,smpB,sucA,tatc,tetrapyrrole_methylase,tgt,Tim44,tmk,tolB,transcriptional_regulator,TraR,trmD,trmE,tRNA_guanine_N7_MT,truA,tryptophanyl_tRNA_synth,ubiA,ubiE,UDP_N_AEPG_reductase,UDP_N_AGPPlase,uvrC,YajC,YidC \
                                --return-best-hit \
                                --get-aa-sequences \
                                --concatenate
                                
# Use trimAL to remove all positions missing from more than 50% of the sequences 

trimal -in concatenated_genes_v2_excl_IV_V/concatenated-best_alphaproteobacteria165.fa \
       -out concatenated_genes_v2_excl_IV_V/concatenated-best_alphaproteobacteria165_out-clean.fa -gt 0.50

# Started with the specific model
iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_504_outgroup_23oct24/concatenated_genes_v2_excl_IV_V/concatenated-best_alphaproteobacteria165_out-clean.fa -m LG+F+R10 -T AUTO -ntmax 25 --alrt 1000 -B 1000 --mem 60G

# Restarted as I realized three somehow remaining IV genomes still in there
iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_504_outgroup_23oct24/concatenated_genes_v2_actually_exl_IV/concatenated-best_alphaproteobacteria165_out-clean_v2.fa -m LG+F+R10 -T AUTO -ntmax 35 --alrt 1000 -B 1000 --mem 60G

```

# Pruned phylogeny (268 SAR11 genomes)

### Generate the external_genomes file without the dereplicated genomes, and other SAGs with ANI values <90% to their closest related genome 

```{r}
library(dplyr)

# Previously the identified IV and V genomes were already removed

# First read in the appropriate old file, this is the most updated with only 484 SAR11 genomes 
external_genomes_484_set_outgroup_v1 <- read.csv("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/data/October24/phylogeny_prep/external_genomes_484_set_outgroup_v1.csv", stringsAsFactors = FALSE)

# Now read in the file with the genomes to retain 
final_SAR11_268 <- read.delim("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/data/October24/phylogeny_prep/final_SAR11_269_name.txt", stringsAsFactors = FALSE)

# Now add the file paths to the final_SAR11_268 file to make a phylogeny
external_genomes_268_set <- left_join(final_SAR11_268, external_genomes_484_set_outgroup_v1, by = c("name"))

write.csv(external_genomes_268_set, file = "/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/data/October24/phylogeny_prep/external_genomes_268_set.csv")

# Now add the outgroup and the path to the outgroup

```

### Make a tree for the 268

```{bash}
# Make an external-genomes.txt file that has all of the genomes and the paths to the contigs.db of interest

cd /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_268_outgroup_29oct24

anvi-get-sequences-for-hmm-hits --external-genomes external_genomes_268_set_outgroup_v1.txt \
                                -o concatenated_genes_268_v1/concatenated-best_alphaproteobacteria165.fa \
                                --hmm-sources Alphaproteobacterial_SCGs \
                                --gene-names acpS,acyl_carrier_protein,afg1_family_ATPase,alpha_beta_hydrolase_fold,ApbC,aspS,ATP_synthase_F0_sub_a,atpG,Beta_lactamase_like_protein,BolA_family_protein,ccmC,ccmE,cell_division_trigger_factor,coaD,coaE,copper_homeostasis,cytidylate_kinase,cytochrome_oxidase_assembly,dapD,delta_AAD,DNA_polymerase_I,DnaA,dnaA,dut_deoxyuridine_TN,eno_hydratase,exonuclease,fabZ,folylpolyglutamate_synth,fpg,frr,gatB,gidA,gidB,glyS,gpsA,grla,GrpE,grxC,hemE,heme_chaperone,Holliday_junction_resolvase_like,HslV,hydrolase_TatD_family,hypothetical_protein_01,hypothetical_protein_02,hypothetical_protein_03,hypothetical_protein_04,hypothetical_protein_05,hypothetical_protein_06,hypothetical_protein_07,hypothetical_protein_08,hypothetical_protein_09,hypothetical_protein_12,hypothetical_protein_14,infC,iojap_protein_family,ksgA,lipoate_protein_ligase_B,lipoyl_synthase,maf,malate_dehydrogenase,mesJ,metallo_beta_lactamase,methylase,miaA,mnmA,MraW,murF,murG,nifR3_TIM,NifU,nucleoside_diphosphate_kinase,nuoJ,nuoK2,nusA,nusG,parvulin_like_PPI,PDT,peptidyl_tRNA_hydrolase,pgk_phosphoglycerate_kinase,pgsA,phage_SPO1_related,pheT,phospho_N_APT,plsX,polyA_polymerase,porphobilinogen_deaminase,protein_S6,putative_DNA_binding,putative_zinc_metallo,pyrG,quinone_oxi,recA,recombination_protein_RecR,Ribonuclease_3,ribonuclease_E,ribosomal_protein_L1,ribosomal_protein_L11,ribosomal_protein_L13,ribosomal_protein_L14,ribosomal_protein_L15,ribosomal_protein_L17,ribosomal_protein_L18,ribosomal_protein_L19,ribosomal_protein_L2,ribosomal_protein_L21,ribosomal_protein_L22,ribosomal_protein_L23,ribosomal_protein_L24,ribosomal_protein_L27,ribosomal_protein_L28,ribosomal_protein_L3,ribosomal_protein_L4,ribosomal_protein_L5,ribosomal_protein_L6,ribosomal_protein_L9,ribosomal_protein_S10,ribosomal_protein_S11,ribosomal_protein_S12,ribosomal_protein_S13,ribosomal_protein_S14,ribosomal_protein_S15,ribosomal_protein_S16,ribosomal_protein_S17,ribosomal_protein_S18,ribosomal_protein_S19,ribosomal_protein_S2,ribosomal_protein_S20,ribosomal_protein_S3,ribosomal_protein_S4,ribosomal_protein_S5,ribosomal_protein_S7,ribosomal_protein_S8,ribosomal_protein_S9,Ribosome_binding_factor_A,RNA_polymerase_sub_a,rnhB,rplT,rrm1,rRNA_LS_methyltransferase,sdhC_556_sub,secY,serS,smpB,sucA,tatc,tetrapyrrole_methylase,tgt,Tim44,tmk,tolB,transcriptional_regulator,TraR,trmD,trmE,tRNA_guanine_N7_MT,truA,tryptophanyl_tRNA_synth,ubiA,ubiE,UDP_N_AEPG_reductase,UDP_N_AGPPlase,uvrC,YajC,YidC \
                                --return-best-hit \
                                --get-aa-sequences \
                                --concatenate
                                
# Use trimAL to remove all positions that are missing from more than 50% of the sequences 

trimal -in concatenated_genes_268_v1/concatenated-best_alphaproteobacteria165.fa \
       -out concatenated_genes_268_v1/concatenated-best_alphaproteobacteria165_out-clean.fa -gt 0.50

# Run the tree in IQtree
iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_268_outgroup_29oct24/concatenated_genes_268_v1/concatenated-best_alphaproteobacteria165_out-clean.fa -m LG+F+R10 -T AUTO -ntmax 10 --alrt 1000 -B 1000 --mem 60G

```


Fixing the last genome, changing HIMB140 to HIMB1556 
Will be updating the name of the external genomes file to "v2"

```{bash}
# Make an external-genomes.txt file that has all of the genomes and the paths to the contigs.db of interest

cd /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_268_outgroup_12nov24

anvi-get-sequences-for-hmm-hits --external-genomes external_genomes_268_set_outgroup_v2.txt \
                                -o concatenated_genes_268_v2/concatenated-best_alphaproteobacteria165.fa \
                                --hmm-sources Alphaproteobacterial_SCGs \
                                --gene-names acpS,acyl_carrier_protein,afg1_family_ATPase,alpha_beta_hydrolase_fold,ApbC,aspS,ATP_synthase_F0_sub_a,atpG,Beta_lactamase_like_protein,BolA_family_protein,ccmC,ccmE,cell_division_trigger_factor,coaD,coaE,copper_homeostasis,cytidylate_kinase,cytochrome_oxidase_assembly,dapD,delta_AAD,DNA_polymerase_I,DnaA,dnaA,dut_deoxyuridine_TN,eno_hydratase,exonuclease,fabZ,folylpolyglutamate_synth,fpg,frr,gatB,gidA,gidB,glyS,gpsA,grla,GrpE,grxC,hemE,heme_chaperone,Holliday_junction_resolvase_like,HslV,hydrolase_TatD_family,hypothetical_protein_01,hypothetical_protein_02,hypothetical_protein_03,hypothetical_protein_04,hypothetical_protein_05,hypothetical_protein_06,hypothetical_protein_07,hypothetical_protein_08,hypothetical_protein_09,hypothetical_protein_12,hypothetical_protein_14,infC,iojap_protein_family,ksgA,lipoate_protein_ligase_B,lipoyl_synthase,maf,malate_dehydrogenase,mesJ,metallo_beta_lactamase,methylase,miaA,mnmA,MraW,murF,murG,nifR3_TIM,NifU,nucleoside_diphosphate_kinase,nuoJ,nuoK2,nusA,nusG,parvulin_like_PPI,PDT,peptidyl_tRNA_hydrolase,pgk_phosphoglycerate_kinase,pgsA,phage_SPO1_related,pheT,phospho_N_APT,plsX,polyA_polymerase,porphobilinogen_deaminase,protein_S6,putative_DNA_binding,putative_zinc_metallo,pyrG,quinone_oxi,recA,recombination_protein_RecR,Ribonuclease_3,ribonuclease_E,ribosomal_protein_L1,ribosomal_protein_L11,ribosomal_protein_L13,ribosomal_protein_L14,ribosomal_protein_L15,ribosomal_protein_L17,ribosomal_protein_L18,ribosomal_protein_L19,ribosomal_protein_L2,ribosomal_protein_L21,ribosomal_protein_L22,ribosomal_protein_L23,ribosomal_protein_L24,ribosomal_protein_L27,ribosomal_protein_L28,ribosomal_protein_L3,ribosomal_protein_L4,ribosomal_protein_L5,ribosomal_protein_L6,ribosomal_protein_L9,ribosomal_protein_S10,ribosomal_protein_S11,ribosomal_protein_S12,ribosomal_protein_S13,ribosomal_protein_S14,ribosomal_protein_S15,ribosomal_protein_S16,ribosomal_protein_S17,ribosomal_protein_S18,ribosomal_protein_S19,ribosomal_protein_S2,ribosomal_protein_S20,ribosomal_protein_S3,ribosomal_protein_S4,ribosomal_protein_S5,ribosomal_protein_S7,ribosomal_protein_S8,ribosomal_protein_S9,Ribosome_binding_factor_A,RNA_polymerase_sub_a,rnhB,rplT,rrm1,rRNA_LS_methyltransferase,sdhC_556_sub,secY,serS,smpB,sucA,tatc,tetrapyrrole_methylase,tgt,Tim44,tmk,tolB,transcriptional_regulator,TraR,trmD,trmE,tRNA_guanine_N7_MT,truA,tryptophanyl_tRNA_synth,ubiA,ubiE,UDP_N_AEPG_reductase,UDP_N_AGPPlase,uvrC,YajC,YidC \
                                --return-best-hit \
                                --get-aa-sequences \
                                --concatenate

# Use trimAL to remove all positions that are missing from more than 50% of the sequences 

trimal -in concatenated_genes_268_v2/concatenated-best_alphaproteobacteria165.fa \
       -out concatenated_genes_268_v2/concatenated-best_alphaproteobacteria165_out-clean.fa -gt 0.50

# Restart the tree
iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_268_outgroup_12nov24/concatenated_genes_268_v2/concatenated-best_alphaproteobacteria165_out-clean.fa -m LG+F+R10 -T AUTO -ntmax 35 --alrt 1000 -B 1000 --mem 60G

```


# Visualize phylogenies in anvi'o


```{r setup, include=FALSE}

library(tidyr)

SAR11_CLUSTER_REPORT_504_v6 <- read.csv("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/data/October24/fastANI_dereplication_prep/SAR11_CLUSTER_REPORT_504_v6.csv", stringsAsFactors = FALSE)

CLUSTER_REPORT_504_v6_long <- separate_longer_delim(SAR11_CLUSTER_REPORT_504_v6,"genomes", delim=";")

write.csv(CLUSTER_REPORT_504_v6_long, file="/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/data/October24/fastANI_dereplication_prep/SAR11_CLUSTER_REPORT_504_v6_long.csv")

```

### 481 genomes

```{bash}

# Time to make a tree, this is for the 481 SAR11 genomes + 10 isolate genomes as an outgroup 
#       -p is for an imaginary profile database
#       -d is to the file with layers
#       -t is to the phylogeny of interest 

# This is for the mega tree that was REDONE yet again since a few IV genomes had been left in there *smacks face* 
cd /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_504_outgroup_23oct24/concatenated_genes_v2_actually_exl_IV

anvi-interactive -p SAR11_481_v2.db \
                 -d LAYERS/LAYERS_SAR11_481_v6.txt \
                 -t concatenated-best_alphaproteobacteria165_out-clean_v2.fa.contree \
                 --title "SAR11_481_v2" \
                 --manual
                 
anvi-import-collection -C collection_genera \
                        -p SAR11_481_v2.db \
                        --bins-info LAYERS/bin_color_info_v2.txt \
                        LAYERS/collection_genera.txt


```


### 268 genomes

Prepare the layers file for the succint tree

```{r}

# import the layers from the 481
LAYERS_SAR11_481_v1 <- read.delim("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/preprint/phylogenies/SAR11_165_504_outgroup_23oct24/concatenated_genes_v2_excl_IV_V/LAYERS/LAYERS_SAR11_481_v1.txt", stringsAsFactors = FALSE)

# now import the list of the 268 
set_268 <- read.delim("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/preprint/phylogenies/SAR11_165_268_outgroup_29oct24/concatenated_genes_268_v1/LAYERS/genomes_268_set.txt", stringsAsFactors = FALSE)

# JOIN these sweet baby angels

LAYERS_SAR11_268_v1 <- left_join(set_268, LAYERS_SAR11_481_v1, by = c("genome"))

# write the file
write_delim(LAYERS_SAR11_268_v1, file="/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/preprint/phylogenies/SAR11_165_268_outgroup_29oct24/concatenated_genes_268_v1/LAYERS/LAYERS_SAR11_268_v1.txt")

# At this point, sort the table by subgroup and check a type genome has been chosen for each one 
# Also finalize the number of genera defined here

```
 
 
```{bash}

# This is for the pruned tree 

anvi-interactive -p SAR11_268_v1.db \
                 -d LAYERS/layers_test_v3.txt \
                 -t concatenated-best_alphaproteobacteria165_out-clean.fa.contree \
                 --title "SAR11_268_v1" \
                 --manual
                 
anvi-import-collection -C collection_genera \
                        -p SAR11_268_v1.db \
                        --bins-info bin_color_info_v1.txt \
                        collection_genera_268_v2.txt



```

# 
```{bash}

# This is for the succinct tree with HIMB1556 vs HIMB140 (nov 12 tree)
cd /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_268_outgroup_12nov24/concatenated_genes_268_v2

anvi-interactive -p SAR11_268_v2.db \
                 -d LAYERS/LAYERS_SAR11_268_v4.txt \
                 -t concatenated-best_alphaproteobacteria165_out-clean.fa.contree \
                 --title "SAR11_268_v2" \
                 --manual
                 
anvi-import-collection -C collection_genera \
                        -p SAR11_268_v2.db \
                        --bins-info bin_color_info_v2.txt \
                        collection_genera_v1.txt
                        
                        
        # Adding a new column that is 'old genome, old HIMB genome or new HIMB genome' for the final phylogeny

anvi-interactive -p SAR11_268_v2.db \
                 -d LAYERS/LAYERS_SAR11_268_v5.txt \
                 -t concatenated-best_alphaproteobacteria165_out-clean.fa.contree \
                 --title "SAR11_268_v2" \
                 --manual     

## Adding detection values to re-make a figure for supplemental reasons   
# mkdir /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_268_outgroup_12nov24/concatenated_genes_268_v2_avedetection
# put relevant files in there

cd /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_268_outgroup_12nov24/concatenated_genes_268_v2_avedetection

anvi-interactive -p SAR11_268_v2.db \
                 -d LAYERS/LAYERS_SAR11_268_v5_averages.txt \
                 -t concatenated-best_alphaproteobacteria165_out-clean.fa.contree \
                 --title "SAR11_268_v2_avs" \
                 --manual    


```

```{r}

# Okay now need to make a tree that has all 481 genomes plus the GTDB classification as well as the RED scores at 0.77, etc

layers_gtdb_17sept24_v3 <- read.csv("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/data/velocirappetor/concatenated_genes_v5_nodups/layers_gtdb_17sept24_v3.txt", header = TRUE,  sep = "\t")

LAYERS_SAR11_481_v15_ecogenomics_GTDB <- LAYERS_SAR11_481_v15_ecogenomics %>% 
  left_join(y=layers_gtdb_17sept24_v3, by=c("name" = "genomes"))

write_delim(LAYERS_SAR11_481_v15_ecogenomics_GTDB, file="/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/preprint/phylogenies/SAR11_165_504_outgroup_23oct24/concatenated_genes_v2_actually_exl_IV/LAYERS/LAYERS_SAR11_481_v15_ecogenomics_GTDB.txt")

# Some manual updates and now its v16
LAYERS_SAR11_481_v16_ecogenomics_GTDB <- read.delim("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/preprint/phylogenies/SAR11_165_504_outgroup_23oct24/concatenated_genes_v2_actually_exl_IV/LAYERS/LAYERS_SAR11_481_v16_ecogenomics_GTDB.txt", header = TRUE,  sep = "\t")

# Going to import this into the 268 genomes layers and make a tree with that because that actually makes more sense
LAYERS_SAR11_268_v5 <- read.delim("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/preprint/phylogenies/SAR11_165_268_outgroup_12nov24/LAYERS/LAYERS_SAR11_268_v5.txt", header = TRUE,  sep = "\t")

LAYERS_SAR11_268_v6 <- LAYERS_SAR11_268_v5 %>% 
  left_join(y=LAYERS_SAR11_481_v16_ecogenomics_GTDB, by=c("accession" = "accession"))

write.csv(LAYERS_SAR11_268_v6, file="/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/preprint/phylogenies/SAR11_165_268_outgroup_12nov24/LAYERS/LAYERS_SAR11_268_v6.csv")

# See below for new anvi interactive

```

# For Figure S8, phylogeny with with GTDB info and RED scores

```{bash}

anvi-interactive -p SAR11_268_v2.db \
                 -d LAYERS/LAYERS_SAR11_268_v6.txt \
                 -t concatenated-best_alphaproteobacteria165_out-clean.fa.contree \
                 --title "SAR11_268_v2_GTDB" \
                 --manual   

```


# To demonstrate Ic genomes are in the same monophyletic clade 

```{bash}

# Make an external-genomes.txt file that has all of the genomes and the paths to the contigs.db of interest

cd /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/Ic

anvi-get-sequences-for-hmm-hits --external-genomes external_genomes_Ic_outgroup.txt \
                                -o Ic/concatenated-best_alphaproteobacteria165.fa \
                                --hmm-sources Alphaproteobacterial_SCGs \
                                --gene-names acpS,acyl_carrier_protein,afg1_family_ATPase,alpha_beta_hydrolase_fold,ApbC,aspS,ATP_synthase_F0_sub_a,atpG,Beta_lactamase_like_protein,BolA_family_protein,ccmC,ccmE,cell_division_trigger_factor,coaD,coaE,copper_homeostasis,cytidylate_kinase,cytochrome_oxidase_assembly,dapD,delta_AAD,DNA_polymerase_I,DnaA,dnaA,dut_deoxyuridine_TN,eno_hydratase,exonuclease,fabZ,folylpolyglutamate_synth,fpg,frr,gatB,gidA,gidB,glyS,gpsA,grla,GrpE,grxC,hemE,heme_chaperone,Holliday_junction_resolvase_like,HslV,hydrolase_TatD_family,hypothetical_protein_01,hypothetical_protein_02,hypothetical_protein_03,hypothetical_protein_04,hypothetical_protein_05,hypothetical_protein_06,hypothetical_protein_07,hypothetical_protein_08,hypothetical_protein_09,hypothetical_protein_12,hypothetical_protein_14,infC,iojap_protein_family,ksgA,lipoate_protein_ligase_B,lipoyl_synthase,maf,malate_dehydrogenase,mesJ,metallo_beta_lactamase,methylase,miaA,mnmA,MraW,murF,murG,nifR3_TIM,NifU,nucleoside_diphosphate_kinase,nuoJ,nuoK2,nusA,nusG,parvulin_like_PPI,PDT,peptidyl_tRNA_hydrolase,pgk_phosphoglycerate_kinase,pgsA,phage_SPO1_related,pheT,phospho_N_APT,plsX,polyA_polymerase,porphobilinogen_deaminase,protein_S6,putative_DNA_binding,putative_zinc_metallo,pyrG,quinone_oxi,recA,recombination_protein_RecR,Ribonuclease_3,ribonuclease_E,ribosomal_protein_L1,ribosomal_protein_L11,ribosomal_protein_L13,ribosomal_protein_L14,ribosomal_protein_L15,ribosomal_protein_L17,ribosomal_protein_L18,ribosomal_protein_L19,ribosomal_protein_L2,ribosomal_protein_L21,ribosomal_protein_L22,ribosomal_protein_L23,ribosomal_protein_L24,ribosomal_protein_L27,ribosomal_protein_L28,ribosomal_protein_L3,ribosomal_protein_L4,ribosomal_protein_L5,ribosomal_protein_L6,ribosomal_protein_L9,ribosomal_protein_S10,ribosomal_protein_S11,ribosomal_protein_S12,ribosomal_protein_S13,ribosomal_protein_S14,ribosomal_protein_S15,ribosomal_protein_S16,ribosomal_protein_S17,ribosomal_protein_S18,ribosomal_protein_S19,ribosomal_protein_S2,ribosomal_protein_S20,ribosomal_protein_S3,ribosomal_protein_S4,ribosomal_protein_S5,ribosomal_protein_S7,ribosomal_protein_S8,ribosomal_protein_S9,Ribosome_binding_factor_A,RNA_polymerase_sub_a,rnhB,rplT,rrm1,rRNA_LS_methyltransferase,sdhC_556_sub,secY,serS,smpB,sucA,tatc,tetrapyrrole_methylase,tgt,Tim44,tmk,tolB,transcriptional_regulator,TraR,trmD,trmE,tRNA_guanine_N7_MT,truA,tryptophanyl_tRNA_synth,ubiA,ubiE,UDP_N_AEPG_reductase,UDP_N_AGPPlase,uvrC,YajC,YidC \
                                --return-best-hit \
                                --get-aa-sequences \
                                --concatenate
                                
# Use trimAL to remove all positions missing from more than 50% of the sequences 

trimal -in Ic/concatenated-best_alphaproteobacteria165.fa \
       -out Ic/concatenated-best_alphaproteobacteria165_out-clean.fa -gt 0.50

# Started with the specific model
iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/Ic/concatenated-best_alphaproteobacteria165_out-clean.fa -m LG+F+R10 -T AUTO -ntmax 25 --alrt 1000 -B 1000 --mem 60G

mkdir /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/Ic_WAG

cp /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/Ic/concatenated-best_alphaproteobacteria165_out-clean.fa /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/Ic_WAG/concatenated-best_alphaproteobacteria165_out-clean.fa

iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/Ic_WAG/concatenated-best_alphaproteobacteria165_out-clean.fa -m WAG -T AUTO -ntmax 25 --alrt 1000 -B 1000 --mem 60G

```

# To demonstrate II genomes are in the same monophyletic clade 

```{r}

II_include <- read.delim("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/preprint/phylogenies/Ic_II_SAG_MAG/II_to_include.txt", header = TRUE,  sep = "\t") 

external_paths <- read.delim("/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/preprint/phylogenies/SAR11_165_504_outgroup_23oct24/external_genomes_504_set_outgroup_v3.txt", header = TRUE,  sep = "\t") 

external_genomes_II_include <- II_include %>% 
  left_join(y=external_paths, by=c("name" = "name"))

write.csv(external_genomes_II_include, file = "/Users/kellefreel/Documents/HIMB/Projects/SAR11/Freel_manu_SAR11/preprint/phylogenies/Ic_II_SAG_MAG/external_genomes_II.csv")

```


```{bash}

# Make an external-genomes.txt file that has all of the genomes and the paths to the contigs.db of interest

cd /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG

anvi-get-sequences-for-hmm-hits --external-genomes external_genomes_II.txt \
                                -o II/concatenated-best_alphaproteobacteria165.fa \
                                --hmm-sources Alphaproteobacterial_SCGs \
                                --gene-names acpS,acyl_carrier_protein,afg1_family_ATPase,alpha_beta_hydrolase_fold,ApbC,aspS,ATP_synthase_F0_sub_a,atpG,Beta_lactamase_like_protein,BolA_family_protein,ccmC,ccmE,cell_division_trigger_factor,coaD,coaE,copper_homeostasis,cytidylate_kinase,cytochrome_oxidase_assembly,dapD,delta_AAD,DNA_polymerase_I,DnaA,dnaA,dut_deoxyuridine_TN,eno_hydratase,exonuclease,fabZ,folylpolyglutamate_synth,fpg,frr,gatB,gidA,gidB,glyS,gpsA,grla,GrpE,grxC,hemE,heme_chaperone,Holliday_junction_resolvase_like,HslV,hydrolase_TatD_family,hypothetical_protein_01,hypothetical_protein_02,hypothetical_protein_03,hypothetical_protein_04,hypothetical_protein_05,hypothetical_protein_06,hypothetical_protein_07,hypothetical_protein_08,hypothetical_protein_09,hypothetical_protein_12,hypothetical_protein_14,infC,iojap_protein_family,ksgA,lipoate_protein_ligase_B,lipoyl_synthase,maf,malate_dehydrogenase,mesJ,metallo_beta_lactamase,methylase,miaA,mnmA,MraW,murF,murG,nifR3_TIM,NifU,nucleoside_diphosphate_kinase,nuoJ,nuoK2,nusA,nusG,parvulin_like_PPI,PDT,peptidyl_tRNA_hydrolase,pgk_phosphoglycerate_kinase,pgsA,phage_SPO1_related,pheT,phospho_N_APT,plsX,polyA_polymerase,porphobilinogen_deaminase,protein_S6,putative_DNA_binding,putative_zinc_metallo,pyrG,quinone_oxi,recA,recombination_protein_RecR,Ribonuclease_3,ribonuclease_E,ribosomal_protein_L1,ribosomal_protein_L11,ribosomal_protein_L13,ribosomal_protein_L14,ribosomal_protein_L15,ribosomal_protein_L17,ribosomal_protein_L18,ribosomal_protein_L19,ribosomal_protein_L2,ribosomal_protein_L21,ribosomal_protein_L22,ribosomal_protein_L23,ribosomal_protein_L24,ribosomal_protein_L27,ribosomal_protein_L28,ribosomal_protein_L3,ribosomal_protein_L4,ribosomal_protein_L5,ribosomal_protein_L6,ribosomal_protein_L9,ribosomal_protein_S10,ribosomal_protein_S11,ribosomal_protein_S12,ribosomal_protein_S13,ribosomal_protein_S14,ribosomal_protein_S15,ribosomal_protein_S16,ribosomal_protein_S17,ribosomal_protein_S18,ribosomal_protein_S19,ribosomal_protein_S2,ribosomal_protein_S20,ribosomal_protein_S3,ribosomal_protein_S4,ribosomal_protein_S5,ribosomal_protein_S7,ribosomal_protein_S8,ribosomal_protein_S9,Ribosome_binding_factor_A,RNA_polymerase_sub_a,rnhB,rplT,rrm1,rRNA_LS_methyltransferase,sdhC_556_sub,secY,serS,smpB,sucA,tatc,tetrapyrrole_methylase,tgt,Tim44,tmk,tolB,transcriptional_regulator,TraR,trmD,trmE,tRNA_guanine_N7_MT,truA,tryptophanyl_tRNA_synth,ubiA,ubiE,UDP_N_AEPG_reductase,UDP_N_AGPPlase,uvrC,YajC,YidC \
                                --return-best-hit \
                                --get-aa-sequences \
                                --concatenate
                                
# Use trimAL to remove all positions missing from more than 50% of the sequences 

trimal -in II/concatenated-best_alphaproteobacteria165.fa \
       -out II/concatenated-best_alphaproteobacteria165_out-clean.fa -gt 0.50

# Started with the specific model
iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/II/concatenated-best_alphaproteobacteria165_out-clean.fa -m LG+F+R10 -T AUTO -ntmax 25 --alrt 1000 -B 1000 --mem 60G

mkdir /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/II_WAG

cp /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/II/concatenated-best_alphaproteobacteria165_out-clean.fa /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/II_WAG/concatenated-best_alphaproteobacteria165_out-clean.fa

iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/II_WAG/concatenated-best_alphaproteobacteria165_out-clean.fa -m WAG -T AUTO -ntmax 25 --alrt 1000 -B 1000 --mem 60G
```


## To visualize w/ outgroup:

```{bash}

# Make an external-genomes.txt file that has all of the genomes and the paths to the contigs.db of interest

cd /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG

anvi-get-sequences-for-hmm-hits --external-genomes external_genomes_II_outgroup_v2.txt \
                                -o II_outgroup_v2/concatenated-best_alphaproteobacteria165.fa \
                                --hmm-sources Alphaproteobacterial_SCGs \
                                --gene-names acpS,acyl_carrier_protein,afg1_family_ATPase,alpha_beta_hydrolase_fold,ApbC,aspS,ATP_synthase_F0_sub_a,atpG,Beta_lactamase_like_protein,BolA_family_protein,ccmC,ccmE,cell_division_trigger_factor,coaD,coaE,copper_homeostasis,cytidylate_kinase,cytochrome_oxidase_assembly,dapD,delta_AAD,DNA_polymerase_I,DnaA,dnaA,dut_deoxyuridine_TN,eno_hydratase,exonuclease,fabZ,folylpolyglutamate_synth,fpg,frr,gatB,gidA,gidB,glyS,gpsA,grla,GrpE,grxC,hemE,heme_chaperone,Holliday_junction_resolvase_like,HslV,hydrolase_TatD_family,hypothetical_protein_01,hypothetical_protein_02,hypothetical_protein_03,hypothetical_protein_04,hypothetical_protein_05,hypothetical_protein_06,hypothetical_protein_07,hypothetical_protein_08,hypothetical_protein_09,hypothetical_protein_12,hypothetical_protein_14,infC,iojap_protein_family,ksgA,lipoate_protein_ligase_B,lipoyl_synthase,maf,malate_dehydrogenase,mesJ,metallo_beta_lactamase,methylase,miaA,mnmA,MraW,murF,murG,nifR3_TIM,NifU,nucleoside_diphosphate_kinase,nuoJ,nuoK2,nusA,nusG,parvulin_like_PPI,PDT,peptidyl_tRNA_hydrolase,pgk_phosphoglycerate_kinase,pgsA,phage_SPO1_related,pheT,phospho_N_APT,plsX,polyA_polymerase,porphobilinogen_deaminase,protein_S6,putative_DNA_binding,putative_zinc_metallo,pyrG,quinone_oxi,recA,recombination_protein_RecR,Ribonuclease_3,ribonuclease_E,ribosomal_protein_L1,ribosomal_protein_L11,ribosomal_protein_L13,ribosomal_protein_L14,ribosomal_protein_L15,ribosomal_protein_L17,ribosomal_protein_L18,ribosomal_protein_L19,ribosomal_protein_L2,ribosomal_protein_L21,ribosomal_protein_L22,ribosomal_protein_L23,ribosomal_protein_L24,ribosomal_protein_L27,ribosomal_protein_L28,ribosomal_protein_L3,ribosomal_protein_L4,ribosomal_protein_L5,ribosomal_protein_L6,ribosomal_protein_L9,ribosomal_protein_S10,ribosomal_protein_S11,ribosomal_protein_S12,ribosomal_protein_S13,ribosomal_protein_S14,ribosomal_protein_S15,ribosomal_protein_S16,ribosomal_protein_S17,ribosomal_protein_S18,ribosomal_protein_S19,ribosomal_protein_S2,ribosomal_protein_S20,ribosomal_protein_S3,ribosomal_protein_S4,ribosomal_protein_S5,ribosomal_protein_S7,ribosomal_protein_S8,ribosomal_protein_S9,Ribosome_binding_factor_A,RNA_polymerase_sub_a,rnhB,rplT,rrm1,rRNA_LS_methyltransferase,sdhC_556_sub,secY,serS,smpB,sucA,tatc,tetrapyrrole_methylase,tgt,Tim44,tmk,tolB,transcriptional_regulator,TraR,trmD,trmE,tRNA_guanine_N7_MT,truA,tryptophanyl_tRNA_synth,ubiA,ubiE,UDP_N_AEPG_reductase,UDP_N_AGPPlase,uvrC,YajC,YidC \
                                --return-best-hit \
                                --get-aa-sequences \
                                --concatenate
                                
# Use trimAL to remove all positions missing from more than 50% of the sequences 

trimal -in II_outgroup_v2/concatenated-best_alphaproteobacteria165.fa \
       -out II_outgroup_v2/concatenated-best_alphaproteobacteria165_out-clean.fa -gt 0.50

# Started with the specific model
iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/II_outgroup_v2/concatenated-best_alphaproteobacteria165_out-clean.fa -m LG+F+R10 -T 25 --alrt 1000 -B 1000 --mem 60G

mkdir /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/II_outgroup_v2_WAG

cp /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/II_outgroup_v2/concatenated-best_alphaproteobacteria165_out-clean.fa /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/II_outgroup_v2_WAG/concatenated-best_alphaproteobacteria165_out-clean.fa

iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_Ic_II_MAG_SAG/II_outgroup_v2_WAG/II_outgroup_v2_WAGconcatenated-best_alphaproteobacteria165_out-clean.fa -m WAG -T AUTO -ntmax 25 --alrt 1000 -B 1000 --mem 60G
```

SAR11_165_type_Ic_II_III

## Updating w/ outgroup, this is a phylogeny of the type strains plus all genomes from Ic, II, and III:
```{bash}

# Make an external-genomes.txt file that has all of the genomes and the paths to the contigs.db of interest

cd /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_type_Ic_II_III

anvi-get-sequences-for-hmm-hits --external-genomes external_genomes_Ic_II_outgroup.txt \
                                -o type_Ic_II_III_v1/concatenated-best_alphaproteobacteria165.fa \
                                --hmm-sources Alphaproteobacterial_SCGs \
                                --gene-names acpS,acyl_carrier_protein,afg1_family_ATPase,alpha_beta_hydrolase_fold,ApbC,aspS,ATP_synthase_F0_sub_a,atpG,Beta_lactamase_like_protein,BolA_family_protein,ccmC,ccmE,cell_division_trigger_factor,coaD,coaE,copper_homeostasis,cytidylate_kinase,cytochrome_oxidase_assembly,dapD,delta_AAD,DNA_polymerase_I,DnaA,dnaA,dut_deoxyuridine_TN,eno_hydratase,exonuclease,fabZ,folylpolyglutamate_synth,fpg,frr,gatB,gidA,gidB,glyS,gpsA,grla,GrpE,grxC,hemE,heme_chaperone,Holliday_junction_resolvase_like,HslV,hydrolase_TatD_family,hypothetical_protein_01,hypothetical_protein_02,hypothetical_protein_03,hypothetical_protein_04,hypothetical_protein_05,hypothetical_protein_06,hypothetical_protein_07,hypothetical_protein_08,hypothetical_protein_09,hypothetical_protein_12,hypothetical_protein_14,infC,iojap_protein_family,ksgA,lipoate_protein_ligase_B,lipoyl_synthase,maf,malate_dehydrogenase,mesJ,metallo_beta_lactamase,methylase,miaA,mnmA,MraW,murF,murG,nifR3_TIM,NifU,nucleoside_diphosphate_kinase,nuoJ,nuoK2,nusA,nusG,parvulin_like_PPI,PDT,peptidyl_tRNA_hydrolase,pgk_phosphoglycerate_kinase,pgsA,phage_SPO1_related,pheT,phospho_N_APT,plsX,polyA_polymerase,porphobilinogen_deaminase,protein_S6,putative_DNA_binding,putative_zinc_metallo,pyrG,quinone_oxi,recA,recombination_protein_RecR,Ribonuclease_3,ribonuclease_E,ribosomal_protein_L1,ribosomal_protein_L11,ribosomal_protein_L13,ribosomal_protein_L14,ribosomal_protein_L15,ribosomal_protein_L17,ribosomal_protein_L18,ribosomal_protein_L19,ribosomal_protein_L2,ribosomal_protein_L21,ribosomal_protein_L22,ribosomal_protein_L23,ribosomal_protein_L24,ribosomal_protein_L27,ribosomal_protein_L28,ribosomal_protein_L3,ribosomal_protein_L4,ribosomal_protein_L5,ribosomal_protein_L6,ribosomal_protein_L9,ribosomal_protein_S10,ribosomal_protein_S11,ribosomal_protein_S12,ribosomal_protein_S13,ribosomal_protein_S14,ribosomal_protein_S15,ribosomal_protein_S16,ribosomal_protein_S17,ribosomal_protein_S18,ribosomal_protein_S19,ribosomal_protein_S2,ribosomal_protein_S20,ribosomal_protein_S3,ribosomal_protein_S4,ribosomal_protein_S5,ribosomal_protein_S7,ribosomal_protein_S8,ribosomal_protein_S9,Ribosome_binding_factor_A,RNA_polymerase_sub_a,rnhB,rplT,rrm1,rRNA_LS_methyltransferase,sdhC_556_sub,secY,serS,smpB,sucA,tatc,tetrapyrrole_methylase,tgt,Tim44,tmk,tolB,transcriptional_regulator,TraR,trmD,trmE,tRNA_guanine_N7_MT,truA,tryptophanyl_tRNA_synth,ubiA,ubiE,UDP_N_AEPG_reductase,UDP_N_AGPPlase,uvrC,YajC,YidC \
                                --return-best-hit \
                                --get-aa-sequences \
                                --concatenate
                                
# Use trimAL to remove all positions missing from more than 50% of the sequences 

trimal -in type_Ic_II_III_v1//concatenated-best_alphaproteobacteria165.fa \
       -out type_Ic_II_III_v1//concatenated-best_alphaproteobacteria165_out-clean.fa -gt 0.50

# Started with the specific model
iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_type_Ic_II_III/type_Ic_II_III_v1/concatenated-best_alphaproteobacteria165_out-clean.fa -m LG+F+R10 -T AUTO -ntmax 25 --alrt 1000 -B 1000 --mem 60G

# Just to be prepared (out of paranoia) also have the same tree with a general WAG model

mkdir /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_type_Ic_II_III/type_Ic_II_III_v1_WAG

cp /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_type_Ic_II_III/type_Ic_II_III_v1/concatenated-best_alphaproteobacteria165_out-clean.fa /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_type_Ic_II_III/type_Ic_II_III_v1_WAG/concatenated-best_alphaproteobacteria165_out-clean.fa

iqtree -s /tank/kfreel_data/SAR11_genomes/Freel_manu_SAR11/preprint/figures/phylogenies/SAR11_165_type_Ic_II_III/type_Ic_II_III_v1_WAG/concatenated-best_alphaproteobacteria165_out-clean.fa -m WAG -T AUTO -ntmax 25 --alrt 1000 -B 1000 --mem 60G
```

